src/Analyzers/CSharp/Analyzers/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionDiagnosticAnalyzer.cs

```diff
@@ -292,7 +292,7 @@ private static SeparatedSyntaxList<ParameterSyntax> GetParameters(AnonymousFunct
         => expression switch
         {
             AnonymousMethodExpressionSyntax anonymousMethod => anonymousMethod.ParameterList?.Parameters ?? default,
-            SimpleLambdaExpressionSyntax simpleLambda => [simpleLambda.Parameter],
+            SimpleLambdaExpressionSyntax simpleLambda => SyntaxFactory.SeparatedList<ParameterSyntax>().Add(simpleLambda.Parameter),
```

Comment:
Blocking on this. These changes were made 6 months ago after agreement with compiler to create and expose these apis, precisely for use cases like this. Roslyn features are not blocked from using the apis we create.  Instead it's the opposite. We should be dogfooding the language and apis as early as we can to be the first wave of feedback and bulwark against problems.